### PR TITLE
Expose sticky select option to the SettingsRegistry

### PR DESCRIPTION
**Pull request by hultonha**
This PR moves the AZ_CVAR to the SettingsRegistry.

It also includes some tidy-up changes to the manipulator test framework as well

### CODE REVIEW

**Comment by hultonha** [comment]
@tjmichaels there's also a small fix in `CViewportTitleDlg::CheckForCameraSpeedUpdate()` to ensure when opening the camera viewport menu we get the most up to date values from the settings registry (`CheckForCameraSpeedUpdate()` was not called before)

File: Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp [hunk 1]
```diff
@@ -652,7 +652,11 @@ namespace UnitTest
         const auto entity1ScreenPosition = AzFramework::WorldToScreen(m_entity1WorldTranslation, m_cameraState);
 
         // click the entity in the viewport
-        m_actionDispatcher->CameraState(m_cameraState)->MousePosition(entity1ScreenPosition)->MouseLButtonDown()->MouseLButtonUp();
+        m_actionDispatcher->SetStickySelect(false)
+            ->CameraState(m_cameraState)
+            ->MousePosition(entity1ScreenPosition)
+            ->MouseLButtonDown()
+            ->MouseLButtonUp();
```
Comment (AMZN-AlexOteiza) [inline Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp:659]: much better!

File: Code/Editor/EditorPreferencesPageViewportGeneral.h [hunk 2]
```diff
@@ -44,6 +43,7 @@ class CEditorPreferencesPage_ViewportGeneral
         float m_defaultFOV;
         float m_defaultAspectRatio;
         bool m_enableContextMenu;
+        bool m_stickySelect;
```
Comment (AMZN-AlexOteiza) [inline Code/Editor/EditorPreferencesPageViewportGeneral.h:46]: ultra nit: m_enableStickySelect perhaps?

File: Code/Editor/EditorPreferencesPageViewportGeneral.cpp [hunk 3]
```diff
@@ -46,10 +48,7 @@ void CEditorPreferencesPage_ViewportGeneral::Reflect(AZ::SerializeContext& seria
         ->Field("ShowGridGuide", &Display::m_showGridGuide)
         ->Field("DisplayDimensions", &Display::m_displayDimension);
 
-    serialize.Class<MapViewport>()
-        ->Version(1)
-        ->Field("SwapXY", &MapViewport::m_swapXY)
-        ->Field("Resolution", &MapViewport::m_resolution);
+    serialize.Class<MapViewport>()->Version(1)->Field("SwapXY", &MapViewport::m_swapXY)->Field("Resolution", &MapViewport::m_resolution);
```
Comment (AMZN-AlexOteiza) [inline Code/Editor/EditorPreferencesPageViewportGeneral.cpp:51]: this was way more readable before, can we keep it like it was?

**Review by AMZN-AlexOteiza — APPROVED** [review]

File: Code/Editor/EditorPreferencesPageViewportGeneral.h (hunk 2 above)
Comment (hultonha) [reply Code/Editor/EditorPreferencesPageViewportGeneral.h:46]: Hmm I'm not a fan of this as `enable` is a verb and makes much more sense as in the context of a function name as opposed to a variable. We could have it be called `m_stickySelectEnabled` (I'm not that fussed but I can update it (and the other name while I'm here if you think that's a good idea))

File: Code/Editor/EditorPreferencesPageViewportGeneral.cpp (hunk 3 above)
Comment (hultonha) [reply Code/Editor/EditorPreferencesPageViewportGeneral.cpp:51]: I'll put a `clang-format off/on` block around it and revert the styling (I do think autoformatting is just so much simpler for nearly everything though, just accepting _this is the style_ does make life a lot simpler, but I can understand in this context it sometimes helping

Comment (hultonha) [reply Code/Editor/EditorPreferencesPageViewportGeneral.cpp]: Now updated :+1:

File: Code/Editor/EditorPreferencesPageViewportGeneral.h (hunk 2 above)
Comment (hultonha) [reply Code/Editor/EditorPreferencesPageViewportGeneral.h]: Now renamed :+1:

**Review by cgalvan — APPROVED** [review]

**Pull request merged**